analytics/aggregate.go

```diff
@@ -364,14 +402,14 @@ func (f *AnalyticsRecordAggregate) Dimensions() (dimensions []Dimension) {
 	return
 }
 
-func (f *AnalyticsRecordAggregate) AsChange() (newUpdate bson.M) {
+func (f *AnalyticsRecordAggregate) AsChange(dimensions []Dimension) (newUpdate bson.M) {
```

Comment:
@kofoworola why are you modifying the `AsChange` func? it's used on mongo pumps.

Reply:
@tbuchaillot i modified it slightly so that the GraphRecordAggregate can call the function with it's own set of dimensions, since it has extra dimensions beyond the AnalyticsRecordAggregate

Reply:
@kofoworola my point is that we are not using the `AsChange` since it's for Mongo pumps, not SQL. Why do we need to modify it now?

Reply:
Oh @tbuchaillot my apologies, this was included from when i started working on the graph mongo aggregate pump

Reply:
Don't worry @kofoworola 😄 let's remove it from this PR and add it in the graph mongo aggregate one!